### DISCUSSION TITLE
Approach for token uri method

### DISCUSSION THREAD

**Discussion by jsonsivar** [fleekxyz/non-fungible-apps#22] [Architecture] [answered]
@EmperorOrokuSaki @zoruka - this is regarding this comment: https://github.com/fleekxyz/contracts/blob/feat/fleek-erc721/contracts/FleekERC721.sol#L80.  I proposed we do it like this so we can build dynamic URI based on sites db in contract.  Otherwise we would have to keep an external JSON file on IPFS synced with this state. 

Maybe after POC we can benchmark that function and see if it's any performance hit and we can think about ways to improve it.  It is a read function so it doesn't take gas but we can still take a closer look to see.

What other pros are there about taking that approach?

**Comment by jsonsivar** [comment, marked as answer]
We already solved this on POC so we can close this.

**Comment by EmperorOrokuSaki** [comment]
Update: I'm closing this discussion due to inactivity and the last comment that states this is solved.
